### PR TITLE
snap: introduce the instance key field

### PR DESCRIPTION
**Pull request by bboozzoo**

### CODE REVIEW

File: snap/info.go [hunk 1]
```diff
@@ -997,3 +997,28 @@ func DropNick(nick string) string {
 	}
 	return nick
 }
+
+// StoreName splits the instance name and returns the store name of the snap.
+func StoreName(name string) string {
```
Comment (zyga) [inline snap/info.go]: Nitpick: please name the argument `instanceName` (also below)

File: snap/info.go [hunk 2]
```diff
@@ -997,3 +997,28 @@ func DropNick(nick string) string {
...
+	return strings.SplitN(name, "_", 2)[0]
+}
+
+// SplitInstanceName splits the instance name and returns the store name and the
+// instance key.
+func SplitInstanceName(name string) (store string, instanceKey string) {
```
Comment (zyga) [inline snap/info.go]: Same request about `name -> instanceName`. You can probably abbreviate return tuple to `(store, instanceKey string)`

File: snap/info.go [hunk 3]
```diff
@@ -997,3 +997,28 @@ func DropNick(nick string) string {
...
+	store = split[0]
+	if len(split) > 1 {
+		instanceKey = split[1]
+	}
+	return store, instanceKey
+}
+
+// InstanceName takes the store name and the instance key and returns an instance
+// name of the snap.
+func InstanceName(storeName string, instanceKey string) string {
```
Comment (zyga) [inline snap/info.go]: Nice :)

File: snap/info.go [hunk 4]
```diff
@@ -144,6 +144,7 @@ type SideInfo struct {
 // Info provides information about snaps.
 type Info struct {
 	SuggestedName string
+	InstanceKey   string
```
Comment (mvo5) [inline snap/info.go]: When/where is this set?

Comment (bboozzoo) [reply snap/info.go]: It will be set based on data from `overlord/snapstate.SnapState`.

File: snap/info.go (hunk 1 above)
Comment (bboozzoo) [reply snap/info.go]: renamed

File: snap/info.go (hunk 2 above)
Comment (bboozzoo) [reply snap/info.go]: renamed and tweaked the return list

File: snap/info.go [hunk 5]
```diff
@@ -997,3 +997,28 @@ func DropNick(nick string) string {
 	}
 	return nick
 }
+
+// StoreName splits the instance name and returns the store name of the snap.
+func StoreName(instanceName string) string {
+	return strings.SplitN(instanceName, "_", 2)[0]
+}
```
Comment (pedronis) [inline snap/info.go]: shouldn't this use SplitInstanceName

File: snap/info.go (hunk 3 above)
Comment (chipaca) [reply snap/info.go]: could also make it `(storeName, instanceKey string)` like above :-)

File: snap/info_test.go [hunk 6]
```diff
@@ -1074,3 +1074,45 @@ func (s *infoSuite) TestNickname(c *C) {
...
+func (s *infoSuite) TestInstanceStoreName(c *C) {
+	store, key := snap.SplitInstanceName("foo_bar")
+	c.Check(store, Equals, "foo")
+	c.Check(key, Equals, "bar")
+
+	store, key = snap.SplitInstanceName("foo")
+	c.Check(store, Equals, "foo")
+	c.Check(key, Equals, "")
+
+	store, key = snap.SplitInstanceName("_bar")
```
Comment (pedronis) [inline snap/info_test.go]: this and the one after should probably come last with a note that they are not expected

**Pull request merged**